playbooks/roles/monitoring-user-defined-projects/README.md

```diff
@@ -0,0 +1,71 @@
+Enable monitoring for user defined projects 
+===========================================
+
+This role enables monitoring services created in the user defined projects. The playbook performs the following steps to enable user defined project monitoring:
+ - Create a new namespace
+ - Deploy a sample prometheus app
+ - Create service to define the monitoring metrics
+ - Deploy a ServiceMonitor resouce object
+ - Update the cluster-monitoring-config object to enable monitoring for user defined projects
+ - Check the service pods running
+ - Grant the permissions to the external user
+ - Extracting the secret token, bearer access token and thanos querier URL
+ - Run querying command to access the monitoring metrics outside the cluster
+
+
+Requirements
+------------
+
+ - Running OCP 4.x cluster.
+
+
+Role Variables
+--------------
+
+| Variable                               | Required | Default                          |Comments                                                                                                                                                                   |
+|----------------------------------------|----------|----------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| enable_user_defined_project_monitoring | yes      | false                            | Flag to be set to true to enable monitoring-user-defined-project-monitoring role execution                                                                                |
+| user_defined_app_name                  | no       | empty string                     | Name for deploying new a sample app, if empty the default app name *"prometheus-example-app"* will be set to the sample app                                               |
+| user_defined_namespace                 | no       | empty string                     | Name for creating a user defined namespace, if empty a random generated name suffixed with "-namespace" will be set to the namespace e.g. "xyz-namespace"                 |
+| service_monitor_name                   | no       | empty string                     | Name for creating a ServiceMonitor object, if empty a random generated name suffixed with "-service-monitor" will be set to the ServiceMonitor e.g. "xyz-service-monitor" |
```

Comment:
Can you please add this in variables or examples.

Reply:
added to the `examples`